### ISSUE TITLE
TLS support for REST API

### ISSUE THREAD

**Issue by malwaredllc** [malwaredllc/minicache#6]
Currently gRPC supports TLS but the HTTP (Gin REST API endpoints) do not use TLS.

The REST API endpoints should support TLS as well.

Relevant section of code [here](https://github.com/malwaredllc/minicache/blob/main/server/server.go#L106-L130)

**Comment by alvinlee001** [comment]
hi @malwaredllc , i am trying to work on this, is it ur intention to enable mTLS for the gin REST endpoints as well? I managed to start it in TLS but it shows: 

```
2022/05/16 15:39:40 http: TLS handshake error from [::1]:53863: EOF
2022/05/16 15:39:40 http: TLS handshake error from [::1]:53864: tls: client didn't provide a certificate
```

**Comment by malwaredllc** [comment]
Closed by #7
